src/sentry/integrations/utils/metrics.py

```diff
@@ -152,8 +153,11 @@ def record_success(self) -> None:
 
         self._terminate(EventLifecycleOutcome.SUCCESS)
 
-    def record_failure(self, exc: BaseException | None = None) -> None:
-        """Record that the event halted in failure.
+    def record_failure(
+        self, exc: BaseException | None = None, data: dict[str, Any] | None = None
```

Comment:
Oh this ties into my earlier question, looks like we can specify extra data upfront here? If so it might be good to rename this to `extra`/`extra_data` just to keep it consistent with the method call.